### DISCUSSION TITLE
Pull into OpenDTU_VeDirect

### DISCUSSION THREAD

**Discussion by helgeerbe** [berni2288/OpenDTU_PowerLimiter#4] [Ideas]
Hi @berni2288, ich wollte gerade anfangen, den PowerLimiter bei mir zu implementieren. Habe jetzt gesehen, dass du eigentlich schon alles erledigt hast. Hast du was dagegen, wenn ich deinen Code übernehme? 
Ich habe kein Pylontech Battery support, aber du hast ja auch Schwellwerte hinterlegt, das würde passen. Wahrscheinlich würde ich als dritte Option noch das Load Flag vom MPPT nehmen, da ich die Schwellwerte dort hinterlegt habe.

**Comment by berni2288** [comment]
Mach nur :)
Irgendwann würd ich sowieso überlegen meine ganzen changes irgendwo wieder upstream in ein anderes Repo zu pushen mit PR, aber solang ich soviel herumbastel...
Wir können auch gern gemeinsam in deinem repo arbeiten, find nur den Reponamen OpenDTU_VeDirect dann nicht mehr ganz passend 😜

**Reply by helgeerbe** [reply]
Wir können gerne zusammen in meinem Repo basteln. Master Branch für Releases, Develop als funktionsfähiger Entwicklungsbranch. Und dann jeweils Erweiterungen in einzelnen Feature Branches.

Ich kann auch noch warten, bis du eine stabile Version hast. Habe eine ziemlich einfache Lösung in Home Assistant implementiert, die erstaunlich gut läuft. Aber wenn der esp das lokal regelt, wäre es einfach schicker und weniger störanfällig.

Hättest du einen Vorschlag für einen besseren Reponamen?

**Reply by berni2288** [reply]
Ja klingt gut, verschieben wir alles in dein Repo, dann arbeite ich dort in einem Featurebranch weiter.

Hmm:
- OpenDTU-Battery
- ok sollten noch mehr Vorschläge werden aber mir fällt sonst nichts ein 😎

**Reply by berni2288** [reply]
Ich hab übrigens Openhab im Einsatz.

**Comment by helgeerbe** [comment]
Ich bin da auch eher kreativlos. 
- OpenDTU-OnBattery fände ich passabel.

Mit Openhab, habe ich gar keine Erfahrung. Dann passt das aber ganz gut, da wir dann ja mehrere Automatisierungssysteme kennen.

**Comment by berni2288** [comment]
ok dann `OpenDTU-OnBattery` :)

**Comment by helgeerbe** [comment]
Habe dich als Contributer eingeladen. Ich hoffe, du kannst jetzt Branches anlegen.

Zum Testen schlage ich for, dass wir unsere Änderungen auf Development mergen und Master nur noch als Freigabe-Branch nutzen.

**Reply by berni2288** [reply]
ja passt, danke.

**Comment by Adminius** [comment]
ähm, jetzt ist mir klar warum die Rebo umbenannt wurde :) 
ich habe schon einen Fork von PowerLimiter und habe dort ein paar Änderungen gemacht (an vielen Stellen). 
Ich habe als Alternative zu MQTT-PowerMeter noch SDM72/630 Support eingebaut. Das funktioniert schon. Nur die UI ist nicht fertig (bin kein VUE experte). Was mache ich jetzt mit dem Zeug? an PowerLimiter Repo als PR senden? Ich hätte es schon gerne alles in einem Repo.

**Reply by helgeerbe** [reply]
Hi @Adminius, die Idee ist schon, dass wir unsere Ideen zusammenwerfen und alles in einem Repo entwicklen. Bei Vue können wir dir ggf. helfen. @berni2288 will seine Erweiterungen in mein Repo einbringen. Daher macht es dann Sinn, wenn du deine Erweiterungen ebenfalls als PR bei mir einstellst. Allerdings erst, wenn wir die Sachen von Bernie drinnen haben. Ich habe ja auch noch ein wenig rumgebastelt und die MQTT Nachrichten geändert. Dass muss jetzt alles zusammengebracht werden.

**Reply by berni2288** [reply]
@Adminius Keine Sorge, es ändert sich für dich nicht viel.
Das Repo hier wird jetzt einfach archiviert und stattdessen werden alle Änderungen in das Repo OpenDTU_VeDirect gemergt (das jetzt OpenDTU-Battery heißt).

Ich hab den master branch von hier grad in den neuen Branch powerlimiter drüben gepusht.

Mach einfach bei https://github.com/helgeerbe/OpenDTU-OnBattery dann einen PR mit Ziel-Branch "[powerlimiter](https://github.com/helgeerbe/OpenDTU-OnBattery/tree/powerlimiter)" auf.

Später wird der Branch powerlimiter drüben dann in den Branch development und dann irgendwann nach master gemergt.
